distributed/client.py

```diff
@@ -1509,6 +1527,9 @@ async def _close(self, fast=False):
 
             await self.rpc.close()
 
+            for preload in self.preloads:
+                await preload.teardown()
+
```

Comment:
Let's move this to call the `teardown` methods earlier in the client close process. If I look at when preload `teardown` happens on the scheduler

https://github.com/dask/distributed/blob/d0a93e950d70680c41959c517a68ee6fc0ce2138/distributed/scheduler.py#L4155-L4156

and worker

https://github.com/dask/distributed/blob/d0a93e950d70680c41959c517a68ee6fc0ce2138/distributed/worker.py#L1615-L1616

it happens towards the beginning before we actually start shutting things down. This can useful if, for example, we want the client's teardown method to communicate with the scheduler. It will also make things more consistent with what we already have with scheduler and worker preloads.

Reply:
I'll suggest we move call the `teardown` methods after 

```python
self.status = "closing"
```

above